keras2onnx/_builtin.py

```diff
@@ -373,6 +373,35 @@ def _calc_explicit_padding(input_size, output_shape, output_padding, kernel_shap
     return pads
 
 
+@converter_func(TYPES.DepthToSpace)
+def convert_tf_depth_to_space(scope, operator, container):
+    node = operator.raw_operator
+    block_size = node.get_attr('block_size')
+    oopb = OnnxOperatorBuilder(container, scope)
+    if _is_nhwc(node):
+        _, h, w, c = _cal_tensor_shape(node.inputs[0])
```

Comment:
I still need some help, because here I have now set `n = -1` such that it is OK if the batch dimension is unknown. But what if `h` and `w` are also unknown, i.e. their values are `None`. How to then do the reshapes below?

Reply:
Good point here. This line has issue when the tensor shape is unknown (dynamic). Search our code base to find the case when `_cal_tensor_shape` is `None`. We need handle that -- Add a `Shape` op after `node.inputs[0]` to get the dynamic shape, and use `Slice` op to get `h` and `w`, and concatenate with other dimensions to make the `desired_shape`. We have some examples in our code.

Reply:
I had a look at an example and it is not trivial, code quickly becomes unreadable because of all the extra nodes added. Then in that case I reconsidered the use of the actual DepthToSpace node, so I worked on that a bit and made it work with two extra transposes. I've added a test case with unknown tensor sizes as well and that one now also passes :-)